llmfoundry/models/llm_embed/modeling_llm_embed.py

```diff
@@ -0,0 +1,447 @@
+        pretrained_model_name_or_path (Optional[str], optional): Pretrained model name or path. Defaults to None.
+        pretrained_lora_id_or_path (Optional[str], optional): Pretrained LoRA (Low Rank Adaptation) ID or path. Defaults to None.
+        trust_remote_code (bool, optional): Whether to trust remote code. Defaults to False.
+        init_device (str, optional): The initial device. Defaults to 'cpu'.
+        use_flash_attention_2 (bool, optional): Whether to use Flash Attention 2. Defaults to True.
+        use_auth_token (bool, optional): Whether to use an authentication token. Defaults to False.
+        config_overrides (Optional[Dict[str, Any]], optional): Overrides for the model configuration. Defaults to None.
+        load_in_8bit (bool, optional): Whether to load the model in 8-bit mode. Defaults to False.
+        loss_fn (str, optional): The loss function to use (either 'torch_crossentropy' or 'fused_crossentropy'). Defaults to 'fused_crossentropy'.
+        **kwargs (Dict[str, Any]): Additional keyword arguments.
+    """
+
+    def __init__(
+        self,
+        tokenizer: PreTrainedTokenizerBase,
+        contrastive_config: Optional[dict[str, Any]] = None,
+        pretrained_model_name_or_path: Optional[str] = None,
+        pretrained_lora_id_or_path: Optional[str] = None,
+        trust_remote_code: bool = False,
+        init_device: str = 'cpu',
+        use_flash_attention_2: bool = True,
+        use_auth_token: bool = False,
+        config_overrides: Optional[dict[str, Any]] = None,
+        load_in_8bit: bool = False,
+        loss_fn: str = 'fused_crossentropy',
+        **kwargs: dict[str, Any],
+    ):
+
+        self.pretrained_model_name_or_path = pretrained_model_name_or_path
+        self.pretrained_lora_id_or_path = pretrained_lora_id_or_path
+        self.trust_remote_code = trust_remote_code
+        self.init_device = init_device
+        self.use_flash_attention_2 = use_flash_attention_2
+        self.use_auth_token = use_auth_token
+        self.config_overrides = config_overrides
+        self.load_in_8bit = load_in_8bit
+        self.kwargs = kwargs
+        self.is_mpt = False
+
+        contrastive_config = contrastive_config or {}
+        contrastive_config_obj: ContrastiveConfig = om.structured(
+            ContrastiveConfig(**contrastive_config),
+        )
+        if tokenizer.pad_token is None:  # type: ignore
+            tokenizer.pad_token = tokenizer.eos_token
+
+        model = self.construct_model()
+
+        train_metrics: list[Metric] = [
+        ]  # TODO: no train metrics for embedding models yet!
+
+        self.eval_metrics = [
+            ContrastiveEvalLoss(),
+        ]
+
+        super().__init__(
+            model=model,
+            tokenizer=tokenizer,
+            use_logits=False,  # Note: set to False
+            metrics=train_metrics,
+            eval_metrics=self.eval_metrics, # type: ignore
```

Comment:
why is this type ignored?

Reply:
Type ignoring until this makes it into the version of composer foundry uses:
(was in composer release yesterday)
https://github.com/mosaicml/composer/pull/3654